internal/auth/client_kerberos.go

```diff
@@ -348,17 +345,19 @@ func (k *kerberosAuthContext) VerifyMIC(micField []byte, micToken []byte) error
 		return err
 	}
 
-	if field.Request != SSH_MSG_USERAUTH_REQUEST || field.Service != "ssh-connection" || field.Method != "gssapi-with-mic" {
+	if field.Request != internalSsh.SSH_MSG_USERAUTH_REQUEST || field.Service != "ssh-connection" || field.Method != "gssapi-with-mic" {
 		return message.NewMessage(
 			message.EAuthKerberosVerificationFailed,
-			"Verify() returned unverified but no error",
+			"Received MIC packet with unexpected values",
```

Comment:
This error message is not very helpful, we should include the values we received if possible.